### PR TITLE
Add libboost-coroutine(-dev) to rosdep

### PR DESCRIPTION
**Pull request by BartKeulen**
Please add the following dependency to the rosdep database.

## Package name:

- libboost-coroutine
- libboost-coroutine-dev

## Package Upstream Source:

- https://github.com/boostorg/coroutine

## Purpose of using this:

Useful when you only want to install the boost coroutine library and not the full boost library.

### libboost-coroutine
- Debian:
  - Bullseye: https://packages.debian.org/bullseye/libboost-coroutine1.74.0
  - Buster: https://packages.debian.org/buster/libboost-coroutine1.67.0
- Fedora: https://packages.fedoraproject.org/pkgs/boost/boost-coroutine/
- Gentoo: https://packages.gentoo.org/packages/dev-libs/boost
- NixOS: https://search.nixos.org/packages?channel=22.11&show=boost&from=0&size=50&sort=relevance&type=packages&query=boost
- openSUSE: https://software.opensuse.org/package/libboost_coroutine1_66_0
- RHEL: https://pkgs.org/search/?q=boost-coroutine
- Ubuntu:
  - Bionic:  https://packages.ubuntu.com/bionic/libboost-coroutine1.65.1
  - Eoan: https://packages.ubuntu.com/eoan/libboost-coroutine1.67.0 (EOL)
  - Focal: https://packages.ubuntu.com/focal/libboost-coroutine1.71.0
  - Jammy: https://packages.ubuntu.com/jammy/libboost-coroutine1.74.0
  - Xenial: https://packages.ubuntu.com/xenial/libboost-coroutine1.58.0 (EOL)

### libboost-coroutine-dev
- Debian: 
  - Bullseye: https://packages.debian.org/bullseye/libboost-coroutine-dev
  - Buster:  https://packages.debian.org/buster/libboost-coroutine-dev
- Fedora: https://packages.fedoraproject.org/pkgs/boost/boost-devel/
- Gentoo: https://packages.gentoo.org/packages/dev-libs/boost
- NixOS: https://search.nixos.org/packages?channel=22.11&show=boost&from=0&size=50&sort=relevance&type=packages&query=boost
- openSUSE: https://software.opensuse.org/package/libboost_coroutine1_66_0-devel?search_term=libboost_coroutine1_66_0-devel
- RHEL: https://pkgs.org/search/?q=boost-devel
- Ubuntu: https://packages.ubuntu.com/search?suite=all&section=all&arch=any&keywords=libboost-coroutine-dev&searchon=names

### CODE REVIEW

File: rosdep/base.yaml [hunk 1]
```diff
@@ -2589,6 +2589,31 @@ libboost-chrono-dev:
...
+libboost-coroutine:
+  debian:
+    bullseye: [libboost-coroutine1.74.0]
+    buster: [libboost-coroutine1.67.0]
+  fedora: [boost-coroutine]
+  gentoo: [dev-libs/boost]
+  nixos: [boost]
+  openembedded: [boost@openembedded-core]
+  opensuse: [libboost_coroutine1_66_0]
+  rhel: [boost-coroutine]
```
Comment (tfoote) [inline rosdep/base.yaml:2601]:
```suggestion
  rhel:
    '*': [boost-coroutine]
    '7': null
```

Nulling out RHEL 7 as caught by the linters.

**Review by tfoote — DISMISSED** [review]
Thanks for the contribution. Please resolve the RHEL 7 linter error then it lgtm.

**Comment by nuclearsandwich** [comment]
@BartKeulen is this PR something you are still interested in? If not can it be closed?

**Comment by BartKeulen** [comment]
> @BartKeulen is this PR something you are still interested in? If not can it be closed?

Yes, I am still interested. Apologies it took a while.

**Review by emersonknapp — APPROVED** [review]
This looks good now, thanks for the followup

File: rosdep/base.yaml [hunk 2]
```diff
@@ -2589,6 +2589,33 @@ libboost-chrono-dev:
   opensuse: [libboost_chrono1_66_0-devel]
   rhel: [boost-devel]
   ubuntu: [libboost-chrono-dev]
+libboost-coroutine:
+  debian:
```
Comment (emersonknapp) [inline rosdep/base.yaml:2593]:
```suggestion
libboost-coroutine:
  alpine: [boost-dev]
  debian:
```

File: rosdep/base.yaml [hunk 3]
```diff
@@ -2589,6 +2589,33 @@ libboost-chrono-dev:
...
+    '*': [boost-coroutine]
+    '7': null
+  ubuntu:
+    bionic: [libboost-coroutine1.65.1]
+    eoan: [libboost-coroutine1.67.0]
+    focal: [libboost-coroutine1.71.0]
+    jammy: [libboost-coroutine1.74.0]
+    xenial: [libboost-coroutine1.58.0]
+libboost-coroutine-dev:
+  debian: [libboost-coroutine-dev]
```
Comment (emersonknapp) [inline rosdep/base.yaml:2611]:
```suggestion
  alpine: [boost-dev]
  debian: [libboost-coroutine-dev]
```

**Review by emersonknapp — CHANGES_REQUESTED** [review]
One more quick addition for Alpine should make the tests happy

**Review by emersonknapp — APPROVED** [review]

**Review by mjcarroll — APPROVED** [review]

**Pull request merged**